pkg/products/marin3r/reconciler_test.go

```diff
@@ -233,3 +237,220 @@ func strPtr(str string) *string {
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimit",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimit2",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v3",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimitv3",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V3",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+					},
+				})
+			},
+			installation: &integreatlyv1alpha1.RHMI{
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
+				},
+			},
+			want:         integreatlyv1alpha1.PhaseCompleted,
+			testFunction: confirmThatConfigRevisionsV3areThereAndV2areGone,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			serverClient := tt.serverClient()
+
+			phase, err := reconcileEnvoyConfigRevisionsDeletion(context.TODO(), serverClient, *tt.installation)
+
+			if err != nil || phase != tt.want {
+				t.Errorf("Found error and phase not completed")
+			}
+
+			err = tt.testFunction(context.TODO(), serverClient, tt.installation.Spec.NamespacePrefix)
+			if err != nil {
+				t.Errorf("ConfigRevisions not cleaned %s", err)
+			}
+		})
+	}
+}
+
+func confirmThatConfigRevisionsAreCleared(ctx context.Context, client k8sclient.Client, nsPrefix string) error {
+	envoyConfigRevisions := &v1alpha1.EnvoyConfigRevisionList{}
+	err := client.List(ctx, envoyConfigRevisions)
+	if err != nil {
+		return err
+	}
+
+	for _, envoyConfigRevision := range envoyConfigRevisions.Items {
+		if envoyConfigRevision.DeletionTimestamp != nil && envoyConfigRevision.Finalizers != nil {
+			return fmt.Errorf("deletion timestamp is present and finalizer is present")
+		}
+
+		if envoyConfigRevision.Finalizers != nil {
+			return fmt.Errorf("finalizer has not been cleaned up properly")
+		}
+
+	}
+
+	return nil
+}
+
+func confirmThatConfigRevisionsV3areThereAndV2areGone(ctx context.Context, client k8sclient.Client, nsPrefix string) error {
```

Comment:
```suggestion
func confirmThatConfigRevisionsV3areThereAndV2areGone(ctx context.Context, client k8sclient.Client) error {
```
parameter unused in the function

Reply:
Same as above. Thanks